src/components/mdx/course.tsx

```diff
@@ -0,0 +1,22 @@
+import useSWR from 'swr'
+import {loadCourse} from '@lib/courses'
+import Link from 'next/link'
+
+export default ({slug}) => {
+  const {data} = useSWR(slug, loadCourse)
```

Comment:
this just works so well 😍

dunno how it handles it from the SSR perspective, but widgets don't really matter and can load lazy/async as far as SEO or whatever is concerned.